client/components/close-on-escape/test/index.jsx

```diff
@@ -0,0 +1,75 @@
+/**
+ * External dependencies
+ */
+import { expect } from 'chai';
+import { shallow } from 'enzyme';
+import React from 'react';
+import ReactDom, { unmountComponentAtNode } from 'react-dom';
+import { spy } from 'sinon';
+
+/**
+ * Internal dependencies
+ */
+import useFakeDom from 'test/helpers/use-fake-dom';
+import CloseOnEscape from '../';
+
+const simulateEscapeKeydown = () => (
+	document.dispatchEvent( new window.KeyboardEvent( 'keydown', { keyCode: 27 } ) )
+);
+
+describe( 'CloseOnEscape', () => {
+	describe( 'rendering', () => {
+		it( 'renders nothing', () => {
+			const wrapper = shallow( <CloseOnEscape /> );
+			expect( wrapper.type() ).to.be.a( 'null' );
+		} );
+	} );
+
+	describe( 'escape keydown event', () => {
+		let onEscapeSpy;
+		let nodes;
+
+		useFakeDom();
+
+		before( () => {
+			onEscapeSpy = spy();
+			nodes = {
+				first: document.createElement( 'div' ),
+				second: document.createElement( 'div' ),
+			};
+		} );
+
+		afterEach( () => {
+			ReactDom.unmountComponentAtNode( nodes.first );
+			ReactDom.unmountComponentAtNode( nodes.second );
+		} );
+
+		it( 'calls the `onEscape` method of stacked components in LIFO order on each escape keydown', () => {
+			ReactDom.render(
```

Comment:
I think @gwwar made good point of using `mount` which is always faster than `render`.

Those two test nodes are never attached to DOM, so I don't think we can properly validate how they are going to behave when nested inside DOM with this test. I echo the proposed test setup proposal.